### PR TITLE
data/selinux: allow checking /var/cache/app-info

### PR DESCRIPTION
**Pull request by bboozzoo**
The appstream-metadata interface sets up access to the host's
/var/cache/app-info. When the interface gets connected, both snapd and
snap-update-ns, check the directory. On SELinux systems, that location is
labeled as fwupd_cache_t. Update the policy to allow searching that location.

The denials are:
```
----
time->Wed Jun  3 14:11:06 2020
type=AVC msg=audit(1591186266.798:588): avc:  denied  { getattr } for  pid=1596
 comm="snapd" path="/var/cache/app-info" dev="dm-0" ino=1182576 
 scontext=system_u:system_r:snappy_t:s0
 tcontext=system_u:object_r:fwupd_cache_t:s0
 tclass=dir permissive=1
----
time->Wed Jun  3 14:11:06 2020
type=AVC msg=audit(1591186266.857:589): avc:  denied  { getattr } for  pid=3530
 comm="snap-update-ns" path="/var/lib/snapd/hostfs/var/cache/app-info" dev="dm-0" ino=1182576
 scontext=system_u:system_r:snappy_mount_t:s0
 tcontext=system_u:object_r:fwupd_cache_t:s0
 tclass=dir permissive=1
```

### CODE REVIEW

**Comment by bboozzoo** [comment]
@Conan-Kudo can you take a look at the changes in snapd.spec?

File: packaging/fedora/snapd.spec [hunk 1]
```diff
@@ -508,10 +508,15 @@ sed -e "s/-Bstatic -lseccomp/-Bstatic/g" -i cmd/snap-seccomp/*.go
...
-pushd ./data/selinux
-make SHARE="%{_datadir}" TARGETS="snappy"
-popd
+(
+%if 0%{?rhel} == 7
+    M4PARAM='-D distro_rhel7'
+%endif
+    # Build SELinux module
+    cd ./data/selinux
+    # pass M4PARAM in env so that make can still manipulate it freely
```
Comment (zyga) [inline packaging/fedora/snapd.spec:517]: Perhaps referring to the make override feature would be more explanatory:
https://www.gnu.org/software/make/manual/html_node/Override-Directive.html

Comment (Conan-Kudo) [reply packaging/fedora/snapd.spec:517]: Yes, this would be helpful. 👍

**Comment by Conan-Kudo** [comment]
This is increasingly stupid. That label is annoying and wrong IMO, but it seems to solve the policy problem. @bboozzoo Can you please file a bug report about this? This label implies that appstream consumers in general will be broken...

**Comment by bboozzoo** [comment]
Filed a bug in RHBZ for the labels: https://bugzilla.redhat.com/show_bug.cgi?id=1843881

**Pull request merged**